docs/src/main/asciidoc/sql.adoc

```diff
@@ -83,17 +70,89 @@ NOTE: If you provide your own `spring.datasource.url`, it will be ignored, unles
-The starter also configures credentials for the JDBC connection based on the properties above.
+The starter also configures credentials for the JDBC connection based on the properties below.
 
 The user properties and the properties provided by the `CloudSqlEnvironmentPostProcessor` are then used by https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-sql.html[Spring Boot] to create the `DataSource`.
 You can select the type of connection pool (e.g., Tomcat, HikariCP, etc.) by https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-sql.html#boot-features-connect-to-production-database[adding their dependency to the classpath].
 
 Using the created `DataSource` in conjunction with Spring JDBC provides you with a fully configured and operational `JdbcTemplate` object that you can use to interact with your SQL database.
 You can connect to your database with as little as a database and instance names.
 
-==== Cloud SQL IAM database authentication
+== R2DBC Support
+
+Maven coordinates, using <<getting-started.adoc#bill-of-materials, Spring Cloud GCP BOM>>:
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.google.cloud</groupId>
+    <artifactId>spring-cloud-gcp-starter-sql-r2dbc-mysql</artifactId>
+</dependency>
+<dependency>
+    <groupId>com.google.cloud</groupId>
+    <artifactId>spring-cloud-gcp-starter-sql-r2dbc-postgresql</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source,subs="normal"]
+----
+dependencies {
+    implementation("com.google.cloud:spring-cloud-gcp-starter-sql-r2dbc-mysql")
+    implementation("com.google.cloud:spring-cloud-gcp-starter-sql-r2dbc-postgresql")
+}
+----
+
+=== Prerequisites
+
+In order to use the Spring Boot Starters for Google Cloud SQL, the Google Cloud SQL API must be enabled in your GCP project.
+
+To do that, go to the https://console.cloud.google.com/apis/library[API library page] of the Google Cloud Console, search for "Cloud SQL API", click the first result and enable the API.
+
+NOTE: There are several similar "Cloud SQL" results.
+You must access the "Google Cloud SQL API" one and enable the API from there.
+
+=== Spring Boot Starter for Google Cloud SQL
+
+The Spring Boot Starters for Google Cloud SQL provide an auto-configured https://docs.oracle.com/javaee/7/api/javax/jms/ConnectionFactory.html[`ConnectionFactory`] bean.
+Coupled with Spring R2DBC it produces a
+https://docs.spring.io/spring-data/r2dbc/docs/current/reference/html/#r2dbc.core[`R2dbcEntityTemplate`] object bean that allows for operations such as querying and modifying a database.
+
+[source,java]
+----
+public Flux<String> listUsers() {
+  return template.select(User.class).all().map(user -> user.toString());
+}
+----
+
+You can rely on
+https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-sql.html#boot-features-connect-to-production-database[Spring Boot data source auto-configuration] to configure a `ConnectionFactory` bean.
```

Comment:
The link does not point to a specific feature. Might be worth skipping this sentence, since you already explain `ConnectionFactory` above.